packages/sign_in_with_apple/lib/src/authorization_credential.dart

```diff
@@ -171,15 +158,16 @@ AuthorizationCredentialAppleID parseAuthorizationCredentialAppleIDFromDeeplink(
   }
 
   final user = deeplink.queryParameters.containsKey('user')
-      ? json.decode(deeplink.queryParameters['user']) as Map<String, dynamic>
+      ? json.decode(deeplink.queryParameters['user'] as String)
+          as Map<String, dynamic>
       : null;
-  final name = user != null ? user['name'] as Map<String, dynamic> : null;
+  final name = user != null ? user['name'] as Map<String, dynamic>? : null;
 
   return AuthorizationCredentialAppleID(
     authorizationCode: deeplink.queryParameters['code'],
-    email: user != null ? user['email'] as String : null,
-    givenName: name != null ? name['firstName'] as String : null,
-    familyName: name != null ? name['lastName'] as String : null,
+    email: user != null ? user['email'] as String? : null,
```

Comment:
```suggestion
    email: user?['email'] as String?,
```

should suffice. Similarly for line 169 and 170.